package/timeJump.js

```diff
@@ -12,20 +14,35 @@ module.exports = (origin, mode) => {
   // recursively change state of tree
   function jump(target, coords = []) {
     const originNode = traverseTree(origin.tree, coords);
-
-    // set the state of the origin tree
-    originNode.component.setState(target.state, () => {
-      // iterate through new children once state has been set
-      target.children.forEach((child, i) => {
-        jump(child, coords.concat(i));
+    // set the state of the origin tree if the component is stateful
+    if (originNode.component.setState) {
+      originNode.component.setState(target.state, () => {
+        // iterate through new children once state has been set
+        target.children.forEach((child, i) => {
+          jump(child, coords.concat(i));
+        });
       });
-    });
+    } else {
+      // if component uses hooks, traverse through the memoize tree
+      let current = originNode.component;
+      let index = 0;
+      const hooks = returnState();
+      // while loop through the memoize tree
+      while (current) {
+        current.queue.dispatch(target.state[hooks[index]]);
+        // Reassign the current value
+        current = current.next;
+        index += 2;
+      }
+    }
   }
 
   return target => {
     // setting mode disables setState from posting messages to window
     mode.jumping = true;
     jump(target);
-    mode.jumping = false;
+    setTimeout(() => {
```

Comment:
Based on testing with Yujin, this is suspected to cause duplicate nodes to render on d3 because of the setTimeout